<launch>
 <arg name="sim_robot" default="true" />
 <arg name="targ_x" default="0.3048" />
 <arg name="targ_y" default="0.0" />
 <arg name="targ_z" default="0.0" />
 <arg name="marker_size" default="11.4" />
 <arg name="max_new_marker_error" default="0.08" />
 <arg name="max_track_error" default="0.2" />
 <arg name="cam_image_topic" default="/camera/camera/color/image_raw" />
 <arg name="cam_info_topic" default="/camera/camera/color/camera_info" />
 <arg name="output_frame" default="/camera_color_optical_frame" />
 <arg name="serial_no_camera" default="802212061325" />
 <arg name="rviz" default="true"/>

 <!-- Launch calibration -->
 <node name = "calibration_node" pkg="pick_and_place_support" type="extrinsic_calibration_node" output="screen" required="true" >
    <param name="sim_robot" value="$(arg sim_robot)"/>
    <param name="targ_x" value="$(arg targ_x)"/>
    <param name="targ_y" value="$(arg targ_y)"/>
    <param name="targ_z" value="$(arg targ_z)"/>
</node>

 <!-- Launch camera node-->
<!-- <include file="$(find realsense2_camera)/launch/rs_rgbd.launch" />-->
 <include file="$(find pick_and_place)/launch/bringup_realsense.launch" >
    <arg name="single_camera" value="true" />
    <arg name="serial_no_camera1" value="$(arg serial_no_camera)" />
    <arg name="tf_prefix_camera1" value="camera"/>
    <arg name="camera1" value="camera" />
 </include>

 <!-- Launch AR tag node -->
  <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
      <param name="marker_size"           type="double" value="$(arg marker_size)" />
      <param name="max_new_marker_error"  type="double" value="$(arg max_new_marker_error)" />
      <param name="max_track_error"       type="double" value="$(arg max_track_error)" />
      <param name="output_frame"          type="string" value="$(arg output_frame)" />

      <remap from="camera_image"  to="$(arg cam_image_topic)" />
      <remap from="camera_info"   to="$(arg cam_info_topic)" />
   </node>

 <!-- Bring up robot -->
 <include file="$(find pick_and_place)/launch/bringup_robot.launch" />

 <!-- Launch planner node -->
 <node name="pick_and_place_node" pkg="pick_and_place" type="pick_and_place_node"/>

 <!-- Launch visualization -->
 <node if="$(arg rviz)" pkg="rviz" type="rviz" name="pick_and_place_viz"
    args="-d $(find pick_and_place_support)/config/calibration.rviz" />

</launch>


